app/controllers/api/generic_objects_controller.rb

```diff
@@ -0,0 +1,53 @@
+module Api
+  class GenericObjectsController < BaseController
+    EXCEPTION_ATTRS = %w(generic_object_definition associations).freeze
+
+    before_action :set_additional_attributes
+
+    def create_resource(_type, _id, data)
+      object_def = retrieve_generic_object_definition(data)
+      object_def.create_object(data.except(*EXCEPTION_ATTRS)).tap do |generic_object|
+        add_associations(generic_object, data)
+        generic_object.save!
+      end
+    rescue => err
+      raise BadRequestError, "Failed to create new generic object - #{err}"
+    end
+
+    def edit_resource(type, id, data)
+      resource_search(id, type, collection_class(type)).tap do |generic_object|
+        generic_object.update_attributes!(data.except(*EXCEPTION_ATTRS))
+        add_associations(generic_object, data)
+        generic_object.save!
+      end
+    rescue => err
+      raise BadRequestError, "Failed to update generic object - #{err}"
+    end
+
+    private
+
+    def set_additional_attributes
+      @additional_attributes = %w(property_attributes)
+      return unless params[:property_associations]
+      params[:property_associations].split(',').each do |prop|
```

Comment:
@abellotti wanted to get your thoughts on this approach to returning associations since it's slightly different than we talked about. With `GET /generic_objects/:id?property_associations=vms,services`, and by adding them to the additional attributes, they can then get returned as if they were a normal relationship, ie:
```
{
  "id": "<generic_object_id>",
   ...
  "vms" [ {"id": "vm_id" }, ... ]
  "services": [ ... ]
}
```
thoughts?

Reply:
I like it, I wonder if the parameter shouldn't just be associations=vms,services,... to match the add_associations/remove_associations actions.

Reply:
@abellotti yeah, I like that. will update! 👍